### ISSUE TITLE
Explore the possibility of increasing the size of Core files

### ISSUE THREAD

**Issue by kgersen** [kgersen/Allegiance#74]
For future development it may be useful to explore the possibility of increasing the limits on the size of Cores.

This would enable the addition of further factions where required without necessitating the removal of other factions.

 _Imported from trac ticket #77,  created by notjarvis on 03-22-2010 at 17:30, last modified: 02-24-2012 at 07:23_

**Comment by kgersen** [comment]
Am I correct in figuring that this would increase the pre/def bits from 399 to whatever we set it at?

If so, I think we should probably keep it at 1000 - Or, rather, 0-999. No need to have them excessively large and cumbersome.

We'd also need to give warning because it would invalidate existing cores, IIRC.

_Trac comment by Andon on 07-12-2010 at 08:20_

**Comment by kgersen** [comment]
the pre/def is an ICE thing - it's not the way the game actually works.  The game uses a string to make up a pseudo bit mask which is limitless in size (for tech masks).  The real issue is the fact we are using only 16 bits to mask part types and hardpoints.

http://www.freeallegiance.org/forums/index.php?showtopic=43955
for those that don't have dev zone access to the above URL:

Imago Jul 30 2008, 12:01 AM said:
"I've noticed the latest cores have invalid Part masks, HullType (equipable) partMask maximums and HardPoint (mountable) partMask maximums.
I know we've commented out assert code to get them to work well enough but why not typedef to an int and use 32-bit partMasks "

This "blew over the heads" of Dogbones and even KGJV at the time.

_Trac comment by Imago on 07-12-2010 at 16:42_

**Comment by kgersen** [comment]
For other core changing things, see tickets #137 #138 and #139

If we're going to be changing the cores, might as well get several changes done at once so we don't invalidate all existing cores every release

_Trac comment by Andon on 07-13-2010 at 20:48_

**Comment by kgersen** [comment]
Deferring core changes to R7

_Trac comment by Xynth on 08-03-2010 at 17:16_

**Comment by LordBlacksun** [comment]
Could this be related to a bug that crashes Allegiance, with too many high resolution, textures loaded at 2048 res, when a match countdown starts? (just after pressing the force launch button?)

**Comment by kgersen** [comment]
no. 'core size' mainly refers to techtree bit mask size for instance and other data type limitation in the core format.
it's not related to high resolution & textures.

As stated a few comment above , this changes should be part of something bigger: a complete overhaul of the core structure. low prio for now
